pkg/credentialprovider/aws/aws_credentials.go

```diff
@@ -54,7 +55,7 @@ type ecrProvider struct {
 	getterFactory tokenGetterFactory
 }
 
-var _ credentialprovider.DockerConfigProvider = &ecrProvider{}
+var _ credentialconfig.DockerConfigProvider = &ecrProvider{}
```

Comment:
I'm not sure that `DockerConfigProvider` really belongs in `k8s.io/cloud-provider`. It is very kubelet specific and an internal implementation detail used by the keyring.

Reply:
Same comment for `DockerConfig` and `DockerConfigJSON` most of the types in `config.go` and `provider.go`.

Reply:
Maybe `k8s.io/kubelet` would suit those types better?

Reply:
On the other hand, I think there's a decent case to be made for `k8s.io/cloud-provider` - the `credentialconfig` package represents the minimum amount of code needed to move the current existing credential provision code into `staging/` without hanging a dependency on anything in `pkg/` afterwards. Since out-of-tree cloud providers will end up using that same code - so both the new plugins as well as existing credential code can benefit from the same shared codebase -  I think `cloud-provider` might make a decent home.

Reply:
I think the conclusion we came to on Slack is that we can put this in a tracking issue and move ahead with this PR for 1.20, so I'm marking this as resolved for the time being, but if anyone has any concerns about that course of action, please feel free to start a conversation again.